### PR TITLE
Add feature specs for better integration coverage of web app

### PR DESCRIPTION
**Pull request by zoso10**
i decided to target one of the most important endpoints first which is the `/entry` endpoint that the gem uses to submit challenge entries. i noticed it was basically a json endpoint so it's best not to use capybara to test it. I used a request spec which is most common for api endpoints.

_note_: ~i'll be breaking up the feature specs into a number of PRs to make them more digestible and easier to review.~ they're broken up into digestible commits. i decided i didn't want to have to deal with rebasing, _but_ upon request i can close this one and make separate PRs if people wish. also, once #186 is merged ill change this to use factories.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -9,6 +9,7 @@
 # Requires supporting ruby files with custom matchers and macros, etc,
 # in spec/support/ and its subdirectories.
 Dir[Rails.root.join("spec/support/**/*.rb")].each {|f| require f}
+Capybara.default_driver = :rack_test
```
Comment (hvenables) [inline spec/spec_helper.rb]: Sure rack test is fine for now, if we ever need a driver that can execute javascript poltergeist is my preferred driver

Comment (zoso10) [inline spec/spec_helper.rb]: agreed, i always have bad luck with webkit and getting qt to play nicely together and selenium seems heavyweight.
there's really not a ton of js in the app right now anyway, the twitter feed integration is the biggest thing. we could probably test that separately anyway.

**Comment by zoso10** [comment]
@hvenables there's a few more specs id like to add, but i think this covers critical paths. i feel a lot more comfortable moving on upgrading, thoughts?

**Comment by hvenables** [comment]
@zoso10 Agreed, Im more comfortable but i would prefer the coverage to be a little higher before starting on upgrading.

**Comment by igrigorik** [comment]
p.s... awesome work guys!

**Pull request merged**